src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -578,100 +416,39 @@ private IEnumerable<(Transaction Tx, Action<Transaction> Change)> UpdateBucket(A
-        {
-            Address address = transaction.SenderAddress;
-            lock (_locker)
-            {
-                if (!_nonces.TryGetValue(address, out var addressNonces))
-                {
-                    addressNonces = new AddressNonces(currentNonce);
-                    _nonces.TryAdd(address, addressNonces);
-                }
-
-                if (!addressNonces.Nonces.TryGetValue(transaction.Nonce, out var nonce))
-                {
-                    nonce = new Nonce(transaction.Nonce);
-                    addressNonces.Nonces.TryAdd(transaction.Nonce, new Nonce(transaction.Nonce));
-                }
-
-                if (!(nonce.TransactionHash is null && nonce.TransactionHash != transaction.Hash))
-                {
-                    // Nonce conflict
-                    if (_logger.IsDebug)
-                        _logger.Debug(
-                            $"Nonce: {nonce.Value} was already used in transaction: '{nonce.TransactionHash}' and cannot be reused by transaction: '{transaction.Hash}'.");
-
-                    return true;
-                }
-
-                nonce.SetTransactionHash(transaction.Hash!);
-            }
-
-            return false;
-        }
-        
         public bool RemoveTransaction(Keccak? hash)
         {
             if (hash is null)
             {
                 return false;
             }
-            
-            bool isKnown;
+
+            bool hasBeenRemoved;
             lock (_locker)
             {
-                isKnown = _transactions.TryRemove(hash, out Transaction transaction);
-                if (isKnown)
+                hasBeenRemoved = _transactions.TryRemove(hash, out Transaction transaction);
+                if (hasBeenRemoved)
                 {
                     Address address = transaction.SenderAddress;
-                    if (_nonces.TryGetValue(address, out AddressNonces addressNonces))
+                    if (_nonces.TryGetValue(address!, out AddressNonces addressNonces))
                     {
                         addressNonces.Nonces.TryRemove(transaction.Nonce, out _);
                         if (addressNonces.Nonces.IsEmpty)
                         {
                             _nonces.Remove(address, out _);
                         }
                     }
-                    
+
                     RemovedPending?.Invoke(this, new TxEventArgs(transaction));
                 }
-                
-                if (_persistentBroadcastTransactions.Count != 0)
-                {
-                    bool ownIncluded = _persistentBroadcastTransactions.TryRemove(hash, out Transaction _);
-                    if (ownIncluded)
-                    {
-                        if (_logger.IsInfo)
-                            _logger.Trace($"Transaction {hash} created on this node was included in the block");
-                    }
-                }
+
+                _broadcaster.StopBroadcast(hash);
```

Comment:
@marcindsobczak I think we need to move that to _txs.Removed event, as evicted transactions might not get removed from broadcasting

Reply:
Or maybe not? Generally what is our broadcasting eviction policy? Is it in line with tx removals? How does it correlate with standard evictions? (Maybe it shouldn't)

Reply:
worth reviewing, it should be slightly cleaner to read now with the broadcaster moved to a separate class